fenix/app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt

```diff
@@ -118,11 +118,22 @@ class InstalledAddonDetailsFragment : Fragment() {
         bindRemoveButton()
     }
 
+    @VisibleForTesting
+    internal fun provideEnableSwitch() = binding.enableSwitch
+
+    @VisibleForTesting
+    internal fun providePrivateBrowsingSwitch() = binding.allowInPrivateBrowsingSwitch
+
+    @VisibleForTesting
     @SuppressWarnings("LongMethod")
-    private fun bindEnableSwitch() {
-        val switch = binding.enableSwitch
-        val privateBrowsingSwitch = binding.allowInPrivateBrowsingSwitch
+    internal fun bindEnableSwitch() {
+        val switch = provideEnableSwitch()
+        val privateBrowsingSwitch = providePrivateBrowsingSwitch()
         switch.setState(addon.isEnabled())
+        if (addon.isDisabledAsBlocklisted()) {
```

Comment:
```suggestion
        // When the ad-on is blocklisted, we do not want to enable the toggle switch
        // because users shouldn't be able to re-enable a blocklisted add-on.
        if (addon.isDisabledAsBlocklisted()) {
```